test/e2e/metrics_test.go

```diff
@@ -77,6 +77,25 @@ func (s *MetricsSuite) TestRetryMetrics() {
 		})
 }
 
+func (s *MetricsSuite) TestFailedMetric() {
+	s.Given().
+		WorkflowTemplate(`@testdata/template-failed-metric.yaml`).
+		Workflow(`@testdata/wf-failed-metric.yaml`).
```

Comment:
is a `WorkflowTemplate` necessary to reproduce this? Or could you just run it as a plain Workflow?

```suggestion
		Workflow(`@testdata/template-failed-metric.yaml`).
```

Reply:
The initial issue was found when using a `WorkflowTemplate`. If you try a metric using `{{status}} == Failed` on the Workflow level, it will still work. Thus this fix is for when the metric is found at the template level.

Reply:
I see what you mean now. I could make that change if you'd want (or make it another case)

Reply:
The tests conventionally use plain Workflows unless a `WorkflowTemplate` is specifically necessary, so I would change to match that

Reply:
Now updated, please re-review when you can, thanks!

Reply:
Small follow-up [below](https://github.com/argoproj/argo-workflows/pull/13163#discussion_r1643076506)